apps/payroll/contracts/Payroll.sol

```diff
@@ -0,0 +1,586 @@
+        tokenContract.approve(address(finance), value);
+        // finally deposit those tokens to Finance
+        finance.deposit(tokenContract, value, "Adding Funds");
+
+        LogFund(
+            from,
+            token,
+            value,
+            tokenContract.balanceOf(this),
+            data
+        );
+
+        return true;
+    }
+
+    /**
+     * @dev Get number of employees in Payroll
+     * @notice Get number of employees in Payroll
+     * @return Number of employees
+     */
+    function getEmployeeCount() public constant auth(OWNER_READ_ROLE) returns(uint256 count) {
+        count = numEmployees;
+    }
+
+    /**
+     * @dev Return all important info too through employees mapping
+     * @notice Return all Employee's important info
+     * @param employeeId Employee's identifier
+     * @return Employee's address to receive payments
+     * @return Employee's annual salary
+     * @return Employee's name
+     * @return Employee's allowed tokens
+     * @return Employee's last call to payment distribution date
+     * @return Employee's last payment received date
+     */
+    function getEmployee(uint256 employeeId)
+        public
+        constant
+        auth(OWNER_READ_ROLE)
+        returns(
+            address accountAddress,
+            uint256 yearlyUSDSalary,
+            string name,
+            address[] allowedTokens,
+            uint lastAllocation,
+            uint lastPayroll
+        )
+    {
+        var employee = employees[employeeId];
+
+        accountAddress = employee.accountAddress;
+        yearlyUSDSalary = employee.yearlyUSDSalary;
+        name = employee.name;
+        allowedTokens = employee.allowedTokensArray;
+        lastAllocation = employee.lastAllocation;
+        lastPayroll = employee.lastPayroll;
+    }
+
+    /**
+     * @dev Get total amount of salaries in Payroll
+     * @notice Get total amount of salaries in Payroll
+     * @return Integer with the amount
+     */
+    function getYearlyTotalPayroll() public constant auth(OWNER_READ_ROLE) returns(uint256 total) {
+        total = yearlyTotalPayroll;
+    }
+
+    /**
+     * @dev Monthly USD amount spent in salaries
+     * @notice Monthly USD amount spent in salaries
+     * @return Integer with the monthly amount
+     */
+    function calculatePayrollBurnrate() public constant auth(OWNER_READ_ROLE) returns(uint256 payrollBurnrate) {
+        payrollBurnrate = yearlyTotalPayroll / 12;
+    }
+
+    /**
+     * @dev Days until the contract can run out of funds
+     * @notice Days until the contract can run out of funds
+     * @return Integer with the number of days
+     */
+    function calculatePayrollRunway() public constant auth(OWNER_READ_ROLE) returns(uint256 payrollRunway) {
+        if (yearlyTotalPayroll == 0)
+            payrollRunway = 2**256 - 1;
+        else
+            payrollRunway = this.balance.mul(365) / yearlyTotalPayroll;
```

Comment:
This wont work now that contract doesnt have any ETH

Reply:
This function should also take token balances into account. It can be a super expensive function gas wise, given that it will be run locally in most instances.

Reply:
Good catch!!